### PR TITLE
Support search & query with dynamic schema

### PR DESCRIPTION
**Pull request by congqixia**
See also milvus-io/milvus#24232
/kind enhancement

### CODE REVIEW

File: client/data.go [hunk 1]
```diff
@@ -76,21 +84,59 @@ func (c *GrpcClient) Search(ctx context.Context, collName string, partitions []s
...
+			}
+			var ok bool
+			dynamicColumn, ok = column.(*entity.ColumnJSONBytes)
+			if !ok {
+				return nil, errors.New("dynamic field not json")
+			}
+		}
+	}
+	columns := make([]entity.Column, 0, len(outputFields))
+	for outputField := range outputs {
```
Comment (yah01) [inline client/data.go:117]: This breaks the order of columns, which may should be aligned to `outputFields`, is it by design?

Comment (yah01) [reply client/data.go:117]: Simply iterate on `outputFields` could solve it

Comment (congqixia) [reply client/data.go]: fixed

**Review by yah01 — APPROVED** [review]
/lgtm

**Pull request merged**